presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoConnectionConfig.java

```diff
@@ -39,26 +49,33 @@
 
     private final HostAndPort address;
     private final URI uri;
+    private final Properties connectionProperties;
 
     private String catalog;
     private String schema;
 
     private final boolean useSecureConnection;
 
-    public PrestoDriverUri(String url)
+    private static final Map<ConnectionProperty, BiConsumer<HttpClientConfig, String>> CLIENT_SETTERS = ImmutableMap.of(
+            SSL_TRUST_STORE_PATH, HttpClientConfig::setTrustStorePath,
+            SSL_TRUST_STORE_PASSWORD, HttpClientConfig::setTrustStorePassword);
+
+    public PrestoConnectionConfig(String url, Properties driverProperties)
             throws SQLException
     {
-        this(parseDriverUrl(url));
+        this(parseDriverUrl(url), driverProperties);
     }
 
-    private PrestoDriverUri(URI uri)
+    private PrestoConnectionConfig(URI uri, Properties driverProperties)
             throws SQLException
     {
         this.uri = requireNonNull(uri, "uri is null");
-        this.address = HostAndPort.fromParts(uri.getHost(), uri.getPort());
+        address = HostAndPort.fromParts(uri.getHost(), uri.getPort());
+        connectionProperties = mergeConnectionProperties(uri, driverProperties);
+
+        validateConnectionProperties(connectionProperties);
 
-        Map<String, String> params = parseParameters(uri.getQuery());
-        useSecureConnection = Boolean.parseBoolean(params.get("secure"));
+        useSecureConnection = SSL.getValue(connectionProperties).get() == SSL_ENABLED;
```

Comment:
For what purpose? I was talking to David on slack, he said it only went in recently and didn't seem opposed to getting rid of it.